src/lexer.coffee

```diff
@@ -807,8 +810,9 @@ exports.Lexer = class Lexer
     low = (codePoint - 0x10000) % 0x400 + 0xDC00
     "#{toUnicodeEscape(high)}#{toUnicodeEscape(low)}"
 
-  # Replace \u{...} with \uxxxx[\uxxxx] in strings and regexes
+  # Replace \u{...} with \uxxxx[\uxxxx] in regexes without `u` flag
   replaceUnicodeCodePointEscapes: (str, options) ->
+    shouldReplace = options.flags? and options.flags.indexOf('u') is -1
```

Comment:
@lydell as currently written, `options` won't always contain a `flags` key (only does if this is for a regex, in which case `flags` will be set to the regex flags string) and `not in` seems to break with an `undefined` arg. Does seem like there should be a simpler way to write this though - I guess you could do something like `options.flags ?= 'u'` before `'u' not in options.flags` to handle the string case but that seems confusing semantically. Or could refactor to pass in eg `dontReplace` option instead of `flags` but it seemed very clean to just pass around existing `flags` and then have the logic for interpreting it with regards to code point escape replacement be contained in this method